modules/aruco/test/test_boarddetection.cpp

```diff
@@ -168,7 +172,7 @@ class CV_ArucoBoardPose : public cvtest::BaseTest {
 CV_ArucoBoardPose::CV_ArucoBoardPose() {}
 
 
-void CV_ArucoBoardPose::run(int) {
+void CV_ArucoBoardPose::run(int run_with) {
```

Comment:
Wrong usage. Don't change semantic of parameters.

Check the base class:

```
    // the main procedure of the test
    virtual void run( int start_from );
```

---

Use derived class or template.

Reply:
removed `run_with` and add ArucoAlgParams to test classes constructor. Also field `Ptr<aruco::DetectorParameters> params` was added to test classes.

Reply:
@alalek, fixed